src/client/caching/stores/MessageStore.ts

```diff
@@ -1,13 +1,78 @@
+/* eslint-disable no-dupe-class-members */
+import { Cache } from '@klasa/cache';
+import { Routes } from '@klasa/rest';
 import { DataStore } from './base/DataStore';
 import { extender } from '../../../util/Extender';
 
-import type { Message } from '../structures/Message';
+import type { APIMessageData } from '@klasa/dapi-types';
 import type { Client } from '../../Client';
+import type { Message } from '../structures/Message';
+import type { TextBasedChannel } from '../../../util/Util';
 
 export class MessageStore extends DataStore<Message> {
 
-	public constructor(client: Client) {
+	public readonly channel: TextBasedChannel;
+
+	public constructor(client: Client, channel: TextBasedChannel) {
 		super(client, extender.get('Message'), client.options.cache.limits.messages);
+		this.channel = channel;
+	}
+
```

Comment:
Missing add/remove methods (especially remove for the purpose of deleting uncached messages).